service/service_test.go

```diff
@@ -0,0 +1,78 @@
+package service
+
+import (
+	. "github.com/smartystreets/goconvey/convey"
+	"os"
+	"testing"
+	"time"
+)
+
+func createMockService() *Service {
+
+	return &Service{
+		Marshaller:    MockMarshaller{},
+		InitialOffset: int64(-1),
+		Schema:        getDefaultSchema(),
+	}
+}
+
+func TestUnitStart(t *testing.T) {
```

Comment:
This missing test coverage for all of the failure scenarios, e.g. failing to unmarshal the Avro.